### PR TITLE
Minor bug fixes to casting

### PR DESCRIPTION
**Pull request by moseleyc**
## Description

- Minor bug fixes to casting: healing should now work, and items costs should be dealt with correctly

## Checklist
- [ ] Tested and working locally
- [ ] Reviewed
- [ ] Tested and working on dev server
- [ ] Deployed to staging

## How to QA
- [x] Inflict damage on a user, heal user with castable
- [x] Cast spell with item cost, user has items, spell is cast and items removed
- [x] Cast spell with item cost, user doesn't have items, spell is not cast and no items removed

## Impacted Areas in Hybrasyl

- Spell casting (Creature / User)

### CODE REVIEW

**Comment by moseleyc** [comment]
#143 Will add to this PR once I get confirmation of how the XSD / XML should look.

**Comment by moseleyc** [comment]
#144 May also be addressed.

**Comment by moseleyc** [comment]
Made one minor change to the Castables XSD, but other than that looks good.

File: HybrasylXML/XSD/Castables.Designer.cs [hunk 1]
```diff
@@ -33,7 +33,7 @@ public partial class Castable
 
         private Categories _categories;
 
-        private CastCostList _castCosts;
```
Comment (cmw-mnorris) [inline HybrasylXML/XSD/Castables.Designer.cs]: This should follow the pattern of List<CastCost> as is part for all other lists. Please do not create new types for items that should be generic.

Comment (cmw-mnorris) [reply HybrasylXML/XSD/Castables.Designer.cs]: and, shit, logged in with my work account, again.

Comment (baughj) [reply HybrasylXML/XSD/Castables.Designer.cs]: No clue what you mean. If you read the diff, the difference is we are making it into a `List<CastCost>` when previously, it was not.

Comment (cmw-mnorris) [reply HybrasylXML/XSD/Castables.Designer.cs]: I read this backwards. Color reversal is a thing, I swear.

**Comment by baughj** [comment]
👍

**Pull request merged**